### PR TITLE
Update napi examples based on latest changes

### PR DESCRIPTION
**Pull request by mhdawson**

### CODE REVIEW

**Comment by mhdawson** [comment]
@nodejs/n-api please review, would like to get landed before Node Interactive

**Comment by mhdawson** [comment]
@sampsongao could you please review.

**Comment by sampsongao** [comment]
LGTM

File: 4_object_factory/napi/addon.cc [hunk 1]
```diff
@@ -19,14 +19,12 @@ napi_value CreateObject(napi_env env, const napi_callback_info info) {
...
-  { name, 0, func, 0, 0, 0, napi_default, 0 }
-
-void Init(napi_env env, napi_value exports, napi_value module, void* priv) {
-  napi_status status;
-  napi_property_descriptor desc = DECLARE_NAPI_METHOD("exports", CreateObject);
-  status = napi_define_properties(env, module, 1, &desc);
+napi_value Init(napi_env env, napi_value exports) {
+  napi_value new_exports;
+  napi_status status =
+      napi_create_function(env, "", -1, CreateObject, nullptr, &new_exports);
```
Comment (sampsongao) [inline 4_object_factory/napi/addon.cc]: this -1 to be NAPI_AUTO_LENGTH

**Comment by mhdawson** [comment]
@sampsongao fixed.

**Review by sampsongao — APPROVED** [review]

**Comment by mhdawson** [comment]
Landed as a1eb8eb8c07d1f88fe3d0dfc42f0063181e31fb9

**Pull request closed without merging**